### PR TITLE
Add demo for box plot, clean up label prop defs for box plot

### PR DESCRIPTION
**Pull request by maddles**

### CODE REVIEW

File: packages/victory-box-plot/src/index.d.ts [hunk 1]
```diff
@@ -14,13 +14,20 @@ import {
...
   StringOrNumberOrCallback,
   VictoryDatableProps,
   VictoryCommonProps,
   VictoryStyleInterface,
   VictoryStyleObject
 } from "victory-core";
 
+export type VictoryBoxPlotLabelType =
+  | boolean
+  | any[]
```
Comment (maddles) [inline packages/victory-box-plot/src/index.d.ts:27]: Wasn't sure about this... should we be enforcing strings and numbers rather than any?

Comment (boygirl) [reply packages/victory-box-plot/src/index.d.ts:27]: I think `victory-core` exposes a `StringOrNumberOrCallback` type that would be useful here

Comment (boygirl) [reply packages/victory-box-plot/src/index.d.ts:27]: Hmm, actually I guess it's a bit different for `VictoryBoxPlot`, but you might be able to borrow some of the ideas from https://github.com/FormidableLabs/victory/blob/master/packages/victory-core/src/index.d.ts#L109

Comment (boygirl) [reply packages/victory-box-plot/src/index.d.ts:27]: And yes, let's use `string | number` instead of any

**Comment by boygirl** [comment]
Oops! Please also run `nps format` and commit the prettier results. That's what's causing the build to fail here.

**Comment by boygirl** [comment]
@maddles this is approved! I'm going to merge now.

**Pull request merged**